### PR TITLE
governance: add threading annotations, add default values, adjust constructors

### PR DESCRIPTION
**Pull request by PastaPastaPasta**

### CODE REVIEW

**Comment by PastaPastaPasta** [comment]
I don't understand these test failures... Help wanted :)

File: src/governance/governance.h [hunk 1]
```diff
@@ -161,38 +161,38 @@ class CGovernanceManager
...
 
-    hash_s_t setRequestedVotes;
+    hash_s_t setRequestedVotes GUARDED_BY(cs);
 
-    bool fRateChecksEnabled;
+    bool fRateChecksEnabled GUARDED_BY(cs) {true};
 
     // used to check for changed voting keys
-    CDeterministicMNListPtr lastMNListForVotingKeys;
+    CDeterministicMNListPtr lastMNListForVotingKeys GUARDED_BY(cs);
```
Comment (pravblockc) [inline src/governance/governance.h:195]: it should be 
CDeterministicMNListPtr lastMNListForVotingKeys GUARDED_BY(cs) {std::make_shared<CDeterministicMNList>()};

Also add below header file as forward declaration is not sufficient
#include <evo/deterministicmns.h>

**Review by UdjinM6 — CHANGES_REQUESTED** [review]
mac build fails https://gitlab.com/dashpay/dash/-/jobs/1631527804

**Comment by PastaPastaPasta** [comment]
We'll see how the mac build likes that. Pay special attention to https://github.com/dashpay/dash/pull/4465/commits/21abbc7aceaecc6b5a6beaa4b661149ba3762c62 (will try to resolve todo in a near future PR), would like your feedback / ensure I'm not crazy when it comes to livecycles / saftey there

**Comment by PastaPastaPasta** [comment]
rebased

**Comment by PastaPastaPasta** [comment]
I figured https://github.com/dashpay/dash/pull/4466/commits/e80b67f5ec047882e8dfc2a5edaf1da644860f25 would resolve these compilation errors, but I guess I'm missing something...

**Review by UdjinM6 — CHANGES_REQUESTED** [review]
mac build complains (werror) + see below

**Pull request closed without merging**